src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -12914,8 +12914,27 @@ private PostSkipAction SkipBadInitializerListTokens<T>(ref SyntaxToken startToke
         {
             return this.SkipBadSeparatedListTokensWithExpectedKind(ref startToken, list,
                 p => p.CurrentToken.Kind != SyntaxKind.CommaToken && !p.IsPossibleExpression(),
-                p => p.CurrentToken.Kind == SyntaxKind.CloseBraceToken || p.IsTerminator(),
+                abortFunc,
                 expected);
+
+            static bool abortFunc(LanguageParser p)
```

Comment:
>abortFunc [](http://example.com/codeflow?start=24&length=9)

The `abortFunc()` local function looks identical here and in `SkipBadArrayInitializerTokens()`. Consider creating a single helper method to reuse across the two callers.